tests/validators/test_url.py

```diff
@@ -30,23 +29,51 @@ def test_url_ok(py_and_json: PyAndJson):
     assert url.port == 443
 
 
+def test_url_ok(py_and_json: PyAndJson):
+    v = py_and_json(core_schema.url_schema())
+    url = v.validate_test('https://example.com/foo/bar?baz=qux#quux')
+
+    assert_example_url(url)
+
+
 def test_url_from_constructor_ok():
     url = Url('https://example.com/foo/bar?baz=qux#quux')
 
-    assert isinstance(url, Url)
-    assert str(url) == 'https://example.com/foo/bar?baz=qux#quux'
-    assert repr(url) == "Url('https://example.com/foo/bar?baz=qux#quux')"
-    assert url.unicode_string() == 'https://example.com/foo/bar?baz=qux#quux'
-    assert url.scheme == 'https'
-    assert url.host == 'example.com'
-    assert url.unicode_host() == 'example.com'
-    assert url.path == '/foo/bar'
-    assert url.query == 'baz=qux'
-    assert url.query_params() == [('baz', 'qux')]
-    assert url.fragment == 'quux'
-    assert url.username is None
-    assert url.password is None
-    assert url.port == 443
+    assert_example_url(url)
+
+
+def test_url_from_build_ok():
+    # 1) no host trailing slash/no path leading slash in the input
+    url = Url.build(scheme='https', host='example.com', path='foo/bar', query='baz=qux', fragment='quux')
+    assert_example_url(url)
+
+    # 2) no host trailing slash/with path leading slash in the input
+    url = Url.build(scheme='https', host='example.com', path='/foo/bar', query='baz=qux', fragment='quux')
+    assert_example_url(url)
+
+    # 3) with host trailing slash/no path leading slash in the input
+    url = Url.build(scheme='https', host='example.com/', path='foo/bar', query='baz=qux', fragment='quux')
+    assert_example_url(url)
+
+    # 4) with host trailing slash/with path leading slash in the input
+    url = Url.build(scheme='https', host='example.com/', path='/foo/bar', query='baz=qux', fragment='quux')
+    assert_example_url(url)
```

Comment:
So I would be tempted to argue that these cases should never be valid, because the `host` isn't valid? Maybe that needs to wait until V3 given it's technically breaking? Or is it kinda broken anyway already because of the unconditional `/` addition?